### PR TITLE
[embedded-elt] Revert #18270 Sling Replication changes.

### PR DESCRIPTION
**Pull request by PedramNavid**
## Summary & Motivation

This reverts #18270 in preparation for adding the new asset decorator approach. I had initially created this revert after the implementation but it made the diff very hard to parse, moving it before the merge makes it a little easier to review the incoming Sling changes.

## How I Tested These Changes
bk

### CODE REVIEW

**Comment by PedramNavid** [comment]
Current dependencies on/for this PR:
* **#19057** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19057?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 4  dependent PRs ([#19274](https://github.com/dagster-io/dagster/pull/19274) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19274?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#19831](https://github.com/dagster-io/dagster/pull/19831) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19831?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#19878](https://github.com/dagster-io/dagster/pull/19878) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19878?utm_source=stack-comment-icon" target="_blank">Graphite</a> and 1 other)
* **#19957** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19957?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19957?utm_source=stack-comment).

**Review by tacastillo — APPROVED** [review]
🫡
